### PR TITLE
Parameter set controller

### PR DESCRIPTION
**Pull request by sgettys**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# target for controller-gen
+generate:
+	mage -v GenerateController
```
Comment (bdegeeter) [inline Makefile]: I put this in as a workaround for controller-gen, but I'm not sure we should keep it.

File: controllers/parameterset_controller.go [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+//+kubebuilder:rbac:groups="",resources=persistentvolumeclaims,verbs=get;list;watch;create;update;patch;delete
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *ParameterSetReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&porterv1.ParameterSet{}, builder.WithPredicates(resourceChanged{})).
+		Owns(&porterv1.AgentAction{}).
+		Complete(r)
+}
+
```
Comment (bdegeeter) [inline controllers/parameterset_controller.go:46]: Missing doc string on public func

Comment (bdegeeter) [inline controllers/parameterset_controller.go:46]: Missing doc string on public func

File: docs/content/file-formats.md [hunk 3]
```diff
@@ -61,6 +62,39 @@ spec:
...
+        secret: test-secret
+```
+
+| Field                     | Required | Default                            | Description                                                 |
+|---------------------------|----------|------------------------------------|-------------------------------------------------------------|
+| agentConfig               | false    | See [Agent Config](#agentconfig)   | Reference to an AgentConfig resource in the same namespace. |
+| porterConfig              | false    | See [Porter Config](#porterconfig) | Reference to a PorterConfig resource in the same namespace. |
+| parameters                | true     |                                    | List of parameter sources for the set |
+| parameters.name           | true     |                                    | The name of the parameter for the bundle |
+| parameters.source         | true     |                                    | The paremeters type. Currently `vaule` and `secret` are the only supported sources |
```
Comment (bdegeeter) [inline docs/content/file-formats.md]: typo

Comment (bdegeeter) [inline docs/content/file-formats.md]: typo

File: docs/content/file-formats.md [hunk 4]
```diff
@@ -61,6 +62,39 @@ spec:
...
+| parameters.source.(value or secret)   | true     |                                    | The plaintext value to use or the name of the secret that holds the parameter |
```
Comment (bdegeeter) [inline docs/content/file-formats.md]: Should `oneOf` be used.  Not sure if json schema aligns with OpenApi on this.

Comment (bdegeeter) [inline docs/content/file-formats.md]: Should `oneOf` be used.  Not sure if json schema aligns with OpenApi on this.

**Review by bdegeeter — CHANGES_REQUESTED** [review]
LGTM, just some minor things to address.

File: docs/content/file-formats.md (hunk 4 above)
Comment (sgettys) [reply docs/content/file-formats.md]: Yeah I wasn't sure what to put here so threw this in. I'll look into OpenAPI spec on how to do this, this was just placeholder

File: Makefile (hunk 1 above)
Comment (sgettys) [reply Makefile]: I think it should stay in, it took me forever how to do this outside of here when I could have just used this!

**Pull request closed without merging**